src/Orleans.Core.Abstractions/IDs/GrainId.cs

```diff
@@ -218,10 +126,12 @@ internal string ToDetailedString()
         private string ToStringImpl(bool detailed)
         {
             string name = string.Empty;
+#if ABSTRACTIONS_TODO
             if (Constants.TryGetSystemGrainName(this, out name))
```

Comment:
if this info is important, you could be passing in the name of the system target in the factory method, and just store it here. We need to consider potential serialization issues with backwards compatibility though